client/devdocs/design/search-collection.jsx

```diff
@@ -112,12 +116,14 @@ const Collection = ( {
 
 	return (
 		<div className="design__collection">
-			{ showCounter > 1 && filter && (
-				<div className="design__instance-links">
-					<span className="design__instance-links-label">Results:</span>
-					{ summary }
-				</div>
-			) }
+			<button onClick={ scroll() } onKeyPress={ scroll() }>
```

Comment:
It seems like the fix is actually working as an unintended result of React rendering. There are two things that need to be fixed here:

First, this button is wrapping the component that renders the search suggestions (once, at the top of the page), not the individual example components on the page, which are set as [an array of `examples` here](https://github.com/Automattic/wp-calypso/pull/44330/files#diff-3d6cda4b838f1af02b0276588d0daf5aL65) and [looped through here](https://github.com/Automattic/wp-calypso/pull/44330/files#diff-3d6cda4b838f1af02b0276588d0daf5aL126). I believe you'd want to wrap [the return here](https://github.com/Automattic/wp-calypso/pull/44330/files#diff-3d6cda4b838f1af02b0276588d0daf5aL104) (or just add the handlers to the wrapper div). You can see where it's rendered by searching for the `design__collection` class in the source.

Second, [the event handling syntax in React is a little different than HTML](https://reactjs.org/docs/handling-events.html). Using `scroll()` (with the parentheses) in a handler in JSX actually calls the function every time the component is rendered. This is why the the patch still "works". Those components are re-rendered when the link in one of the examples is clicked, and thus, `scroll` fires. If you were to add a `console.log( 'scroll' )` to the `scroll` function, you'll see that it fires immediately when the page loads (and any time _any_ element on the screen that causes a re-render is clicked).

Reply:
Thank you for the feedback @michaeldcain!  I made the recommended fixes.  The linter wouldn't allow me to add an event handler to the div tag, so I added the button tag to make it happy.  :smile:  I also added the Fix #24337 to the original commit message as you recommended.  Thank you for your guidance.